src/tests.c

```diff
@@ -408,41 +408,120 @@ void run_scratch_tests(void) {
 /***** HASH TESTS *****/
 
 void run_sha256_tests(void) {
-    static const char *inputs[8] = {
+    static const char *inputs[] = {
         "", "abc", "message digest", "secure hash algorithm", "SHA256 is considered to be safe",
         "abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq",
         "For this sample, this 63-byte string will be used as input data",
-        "This is exactly 64 bytes long, not counting the terminating byte"
+        "This is exactly 64 bytes long, not counting the terminating byte",
+        "aaaaa"
+        /*  Uncomment for test with extremely long input message. */
+        /*, "abcdefghbcdefghicdefghijdefghijkefghijklfghijklmghijklmnhijklmno" */
     };
-    static const unsigned char outputs[8][32] = {
+    static const unsigned int repeat[] = {
+        1, 1, 1, 1, 1, 1, 1, 1, 1000000/5, 16777216
```

Comment:
It's not particularly significant. It's a 1 million bytes test. It's also taken from https://www.di-mgt.com.au/sha_testvectors.html, see the commit message. I left it in because I can't notice a difference of test running time in practice and it won't hurt.

Reply:
The difference is about 3 seconds (10%) in normal, no valgrind, tests.

Reply:
This should use the `CONDITIONAL_TEST` introduced in #1049 .

Reply:
fixed